src/tools/compiletest/src/runtest.rs

```diff
@@ -2544,6 +2545,7 @@ impl<'test> TestCx<'test> {
             }
         }
         if !explicit {
+            let proc_res = self.compile_test(&["--error-format", "json"]);
```

Comment:
Yeah, let's just compile all the tests twice.
Who cares about time, certainly [not Appveyor](https://github.com/rust-lang/rust/issues/46903#issuecomment-357061260).